Dev

Warhammer4000 opened the pull request:
@coderabbitai could you please provide me with a proper changelog for this PR?

@coderabbitai could you please provide me with a proper readme for this PR?

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced a new save system with enhanced encryption, archiving, and multiple slot management.
	- Added a save system prefab for easy integration into projects.
	- Implemented asynchronous save and load operations for improved performance.

- **Enhancements**
	- Improved cloud save functionality and added configuration options for save paths and slot management.
	- Integrated Unity's UniTask for efficient asynchronous operations and added support for file archiving using SharpZipLib utility.

- **Bug Fixes**
	- Updated GUIDs for assets to ensure correct references.

- **Documentation**
	- Provided samples and demo scenes to showcase save system usage.

- **Platform Support**
	- Added new settings for Xbox Series S/X and Xbox One deployment.

- **Refactor**
	- Restructured save system installation and encryption logic for simplicity and reliability.

- **Upgrades**
	- Updated Unity editor version to enhance compatibility and performance.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Warhammer4000 committed:
v2.0.0

#Major Refactoring
- Will provide detailed changelog with AI

Warhammer4000 committed:
v2.0.0

PR initiate

Warhammer4000 commented:
@coderabbitai could you please provide me with a proper changelog for this PR? from all commits

Warhammer4000 commented:
@coderabbitai could you please provide me with a proper readme for this PR from all commits

Warhammer4000 commented:
> @Warhammer4000 The changelog for this PR, based on all commits, is as follows:
> 
> ```
> 71f3fa0 - v2.0.0 (Tanimul Haque Khan, 42 minutes ago)
> 0f2b6d3 - v2.0.0 (Tanimul Haque Khan, 23 hours ago)
> ```
> 
> It appears that there are two commits both marked as version 2.0.0. If you need more detailed information about the changes in each commit, I can provide that as well.
> 
> Analysis chain
> Scripts executed

Please provide a detailed changelog

Warhammer4000 commented:
@coderabbitai please provide a detailed changelog

Warhammer4000 commented:
@coderabbitai generate changelog by comparing code from main and this pull request. don't use commit message as changelog. Instead use file compare.

Warhammer4000 commented:
@coderabbitai please provide further details on specific changes within the files.

Warhammer4000 committed:
## [v2.0.0] - 2024-01-16

### Updated

+ Updated GitHub Actions workflow .github/workflows/UPMBranchUpdate.yml.
+ Modified SaveSystemInstallerWindow.cs script.
+ Updated SaveSystem.cs and SaveSlot.cs.
+ Updated ISaveable.cs interface.
+ Moved FakeData.cs from Tests/PlayMode/Data to Samples/Demo/Scripts and updated metadata.
+ Deleted test scripts BasicSaveLoadTests.cs, EncryptionTest.cs, and SlotMetaDataTest.cs.

### Added

+ Added SlotConfiguration.cs.
+ Added FileProcessor.cs , replacing the deleted SaveProcessor.cs.
+ Added SaveSlotProcessor.cs.
+ Added new samples for demonstration and their configurations.
+ Added Archiver directory and ArchiverBase.cs, ZipUtilityArchiver.cs.
+ Added Encryptor directory and EncryptorBase.cs, AESEncryptor.cs , and deleted Stitcher.cs.
+ Added ItemDataBehaviour.cs and PlayerDataBehaviour.cs scripts for demo.

Warhammer4000 committed:
Fixed sample reference and added editor tool

##ADDED
+ Now save system has editor tools for debug purpose
+ package.json now register sample folder

gitOpu commented on Assets/Packages/com.studio23.ss2.savesystem/Runtime/Core/SaveSlotProcessor.cs:
```diff
@@ -0,0 +1,184 @@
+        {
+            SaveSlot slotMeta=new SaveSlot($"Save Slot {index}");
+
+            string slotMetaPath = Path.Combine(_slotConfiguration.SavePathRoot, $"Save Slot {index}", _slotConfiguration._slotMetafileName);
```
$"Save Slot {index}"
white space may be problematic on different platforms. it's better to remove white space from the path or folder name

Warhammer4000 replied:
Fixed

gitOpu commented on Assets/Packages/com.studio23.ss2.savesystem/Runtime/Data/SlotConfiguration.cs:
```diff
@@ -0,0 +1,23 @@
+
+[CreateAssetMenu(fileName = "SlotConfig", menuName = "Studio-23/SaveSystem/Config", order = 1)]
+public class SlotConfiguration : ScriptableObject
+{
+    [SerializeField] public int _slotCount = 10;
+
+    [SerializeField] public bool _enableBackups = true;
+
+
+    public string _slotDatafolderName = "Data";
+    public string _slotDataBackupFileName = "Data.bak";
+    public string _slotMetafileName = "data.meta";
+
+    public string saveFileExtention = ".taz";
+
+
+    public string _saveRootFolderName = "SaveData";
```
public variable should start in the upper case

Warhammer4000 replied:
Done

gitOpu commented on Assets/Packages/com.studio23.ss2.savesystem/Samples/Demo/Scripts/ItemDataBehaviour.cs:
```diff
@@ -0,0 +1,25 @@
+public class ItemDataBehaviour : MonoBehaviour, ISaveable
+{
+
+    public string _uniqueID;
```
there should be instructions as [Title], for creating uniqueID,

Warhammer4000 replied:
it's from sample.

BDeshiDev commented on Assets/Packages/com.studio23.ss2.savesystem/Runtime/Core/SaveSlotProcessor.cs:
```diff
@@ -0,0 +1,184 @@
+
+        [SerializeField] internal SlotConfiguration _slotConfiguration;
+        [SerializeField] internal FileProcessor _fileProcessor;
+        [SerializeField] internal ArchiverBase archiverBase;
```
Naming convention underscore

Warhammer4000 committed:
Modifications

+ Made abstract classes public so that it can be inherited
+ Save Slot {index} renamed as Save_Slot_{index} for cross platform support
+ Added an editor script to make changes to the configuration file from save system inspector.

BDeshiDev commented on Assets/Packages/com.studio23.ss2.savesystem/Runtime/Data/SlotConfiguration.cs:
```diff
@@ -1,23 +1,26 @@
+        [SerializeField] public int SlotCount = 10;
 
-    [SerializeField] public bool _enableBackups = true;
+        [SerializeField] public bool EnableBackups = true;
```
[SerializeField] not necessary

BDeshiDev commented on Assets/Packages/com.studio23.ss2.savesystem/Editor/Scripts/SaveSlotProcessorEditor.cs:
```diff
@@ -0,0 +1,46 @@
+namespace Studio23.SS2.SaveSystem.Editor
+{
+    [CustomEditor(typeof(SaveSlotProcessor))]
+    public class SaveSlotProcessorEditor : UnityEditor.Editor
+    {
+        private bool showSlotConfig = true;
```
Naming convention.

BDeshiDev commented on Assets/Packages/com.studio23.ss2.savesystem/Editor/Scripts/SaveSystemEditor.cs:
```diff
@@ -1,21 +1,20 @@
     public class SaveSystemEditor : UnityEditor.Editor
     {
         private int slotIndex = 0; // Default slot index
         private bool showDebugTools = true;
```
naming convention

BDeshiDev reviewed (commented):
Minor var name convention changes

Warhammer4000 committed:
Fixed naming conventions
